dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/enrollment/AbstractEnrollmentService.java

```diff
@@ -1201,8 +1200,12 @@ private boolean doValidationOfMandatoryAttributes( User user )
             validateAttributeType( attribute, importOptions, importConflicts );
         }
 
-        TrackedEntityInstance instance = trackedEntityInstanceService.getTrackedEntityInstance( trackedEntityInstance, TrackedEntityInstanceParams.TRUE,
-            importOptions.getUser() );
+        List<String> errors = trackerAccessManager.canRead( importOptions.getUser(), trackedEntityInstance );
+
+        if ( !errors.isEmpty() )
+        {
+            throw new IllegalQueryException( errors.toString() );
```

Comment:
Shouldn't we use ImportSummary here, rather than throw an Exception?

Reply:
Yes, 100%, but it would break how it currently works. The getTrackedEntityinstance called did this IllegalQueryException, and I didn't want to break compatibility (it also broke several tests). Since this is getting backported, I say stay with what we have right now without changing how the API contract behaves. For the new importer this is already gone long time ago.